pandas/core/arrays/timedeltas.py

```diff
@@ -150,6 +151,22 @@ def _box_func(self, x: np.timedelta64) -> Timedelta | NaTType:
             return NaT
         return Timedelta._from_value_and_reso(y, reso=self._creso)
 
+    # error: Signature of "freq" incompatible with supertype "DatetimeLikeArrayMixin"
+    @property  # type: ignore[override]
+    @doc(dtl.DatetimeLikeArrayMixin.freq)
```

Comment:
this should be the same for DatetimeArray too right?  could do it in TimeLikeOps? (similar to but hopefully cleaner than #49952)

Reply:
I will see what mypy says about moving this to `TimeLikeOps`. The doc-string of `DatetimeArray` says that `freq` can be any `(Base)Offset` (not just a `Tick`), so I'm not sure whether it can be moved to `TimeLikeOps`.

It would be nice to have an easier way to keep the same implementation of a `property` but to only change its type annotation.

Reply:
ahh good point.  For TimedeltaArray it is Tick | None, for DTA it is BaseOffset | None